app/views/directives/key-value-editor.html

```diff
@@ -0,0 +1,239 @@
+          <span ng-if="entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip" class="help action-inline">
+            <a
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip}}"
+              title="{{entry.keyValidatorErrorTooltip || keyValidatorErrorTooltip}}">
+              <i class="{{entry.keyValidatorErrorTooltipIcon || keyValidatorErrorTooltipIcon}}"></i>
+            </a>
+          </span>
+        </span>
+        <span
+          class="help-block key-min-length"
+          ng-show="(forms.keyValueEditor[uniqueForKey(unique, $index)].$error.minlength)">
+          <span class="validation-text">Minimum character count is {{keyMinlength}}</span>
+        </span>
+        <span
+          class="help-block key-validation-error"
+          ng-show="(forms.keyValueEditor[uniqueForKey(unique, $index)].$error.required)">
+          <span class="validation-text">{{keyRequiredError}}</span>
+        </span>
+      </div>
+
+      <!-- the value block -->
+      <div
+        class="form-group key-value-editor-input"
+        ng-class="forms.keyValueEditor[uniqueForValue(unique, $index)].$invalid ? 'has-error' : ''">
+
+        <label for="uniqueForValue(unique, $index)" class="sr-only">{{valuePlaceholder}}</label>
+
+        <div ng-if="(!entry.valueFrom)">
+          <input
+            ng-if="(!entry.valueAlt)"
+            type="text"
+            class="form-control"
+            ng-class="{ '{{setFocusValClass}}' : $last  }"
+            id="{{uniqueForValue(unique, $index)}}"
+            name="{{uniqueForValue(unique, $index)}}"
+            ng-attr-placeholder="{{ (!isReadonlyAny) && valuePlaceholder || ''}}"
+            ng-minlength="{{valueMinlength}}"
+            maxlength="{{valueMaxlength}}"
+            ng-model="entry.value"
+            ng-readonly="isReadonlyAny || isReadonlySome(entry.name) || entry.isReadonly"
+            ng-pattern="validation.val"
+            ng-required="!allowEmptyKeys && entry.value">
+        </div>
+
+        <div ng-if="entry.valueFrom">
+          <div ng-if="isReadonlyAny || entry.isReadonlyValue" class="faux-input-group">
+            <span
+              class="faux-form-control-addon {{entry.valueIcon}}"
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.valueIconTooltip || valueIconTooltip}}"
+              title="{{entry.valueIconTooltip || valueIconTooltip}}"></span>
+            <div
+              class="faux-form-control readonly">
+              <span ng-switch="entry.refType">
+                <span ng-switch-when="configMapKeyRef">
+                  Set to the key {{entry.valueFrom.configMapKeyRef.key}} in config map
+                  <span
+                    ng-if="entry.isReadonlyValue">
```

Comment:
why is the appearance of the name as a link dependent on whether the value is readonly?  shouldn't the appearance of the link depend on whether the user has rights to view secrets / view config maps using a canI check?